packages/web-client/src/helpers/share/role.ts

```diff
@@ -356,54 +356,50 @@ export abstract class PeopleShareRoles {
 }
 
 export abstract class LinkShareRoles {
-  static readonly all = [
-    linkRoleViewerFile,
-    linkRoleViewerFolder,
-    linkRoleContributorFolder,
-    linkRoleEditorFolder,
-    linkRoleUploaderFolder
-  ]
-
   static list(
     isFolder: boolean,
     canEditFile = false,
+    canContribute = false,
     hasAliasLinks = false,
     hasPassword = false
   ): ShareRole[] {
     return [
       ...(hasAliasLinks && !hasPassword ? [linkRoleInternalFile, linkRoleInternalFolder] : []),
-      ...this.all,
+      linkRoleViewerFile,
+      linkRoleViewerFolder,
+      ...(canContribute ? [linkRoleContributorFolder] : []),
+      linkRoleEditorFolder,
+      linkRoleUploaderFolder,
       ...(canEditFile ? [linkRoleEditorFile] : [])
     ].filter((r) => r.folder === isFolder)
   }
 
   static getByBitmask(bitmask: number, isFolder: boolean): ShareRole {
-    return [...this.all, linkRoleEditorFile, linkRoleInternalFile, linkRoleInternalFolder] // Always return all roles
-      .find((r) => r.folder === isFolder && r.bitmask(false) === bitmask)
+    return this.list(isFolder, true, true, true, false).find((r) => r.bitmask(false) === bitmask)
```

Comment:
This means 2 finds, but the list is small.. Let me know if you prefer this more optimized.

Reply:
that's fine. I prefer the readability over the performance in this case (like you said, only few items in the list).